bin/deploy_automation/_shared_functions.sh

```diff
@@ -99,4 +99,20 @@ _assert_current_branch_is_not() {
                          "Please, checkout a different branch (other than [${branch_names[@]}])."
     fi
   done
-}
\ No newline at end of file
+}
+
+_git_fetch_and_cleanup() {
+  # fetch from remote origin
+  git fetch origin
+
+  # remove any dangling version tags not present in the remote origin
+  # (this could happen if a previous run failed to push tags to remote)
+  git tag -d $(
+    grep -vxFf  \
+      <(git ls-remote --tags origin | cut -f 2 | sed -E 's/^refs\/tags\///') \
+      <(git tag -l \
+          | grep -E '^v(?:[0-9]+\.){1,2}(?:[0-9]+)_[^_]+_[0-9]{4}-[0-9]{2}-[0-9]{2}$'
+      ) \
+      || [[ $? == 1 ]]
```

Comment:
`grep` exits with code 1 when no lines are found (and other error codes if something else goes wrong). This condition tells bash exit code 1 is actually fine for this command.